### PR TITLE
Polyhedron_Demo: Import AABB_Tree demo

### LINKED ISSUE #1228: Item not cut
When loading an polyhedron item while the cutting plane exists, it is not cut by the plane, we need to move it first so that the cut edges are upated.

### PR DESCRIPTION
**Pull request by maxGimeno**
This PR adds distance functions visualization to the Cut_plugin. This feature is taken from the AABB_Tree demo.
It also fixes #1228 .

### CODE REVIEW

File: Polyhedron/demo/Polyhedron/Scene_polyhedron_item.cpp [hunk 1]
```diff
@@ -1753,3 +1753,8 @@ bool Scene_polyhedron_item::triangulated(){return d->poly->is_pure_triangle();}
 bool Scene_polyhedron_item::self_intersected(){return !(d->self_intersect);}
 void Scene_polyhedron_item::setItemIsMulticolor(bool b){ d->is_multicolor = b;}
 bool Scene_polyhedron_item::isItemMulticolor(){ return d->is_multicolor;}
+
+void* Scene_polyhedron_item::get_aabb_tree()
+{
+  return d->get_aabb_tree();
+}
```
Comment (lrineau) [inline Polyhedron/demo/Polyhedron/Scene_polyhedron_item.cpp]: Please remove this function from the pull-request. It is not used anywhere.

File: Polyhedron/demo/Polyhedron/Scene_polyhedron_item.h [hunk 2]
```diff
@@ -62,7 +62,6 @@ class SCENE_POLYHEDRON_ITEM_EXPORT Scene_polyhedron_item
     bool load_obj(std::istream& in);
     bool save(std::ostream& out) const;
     bool save_obj(std::ostream& out) const;
-
```
Comment (lrineau) [inline Polyhedron/demo/Polyhedron/Scene_polyhedron_item.h]: If you re-add that empty line in the file, then the `.h` will not be modified, and that will avoid to recompile all plugins that depend on the polyhedron item.

**Comment by sloriot** [comment]
link error `CMakeFiles/cut_plugin.dir/Cut_plugin.cpp.o: In function `Polyhedron_demo_cut_plugin::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)':
/home/sloriot/CGAL/git/integration/build/test-g++/demo/Polyhedron/Plugins/AABB_tree/Cut_plugin.moc:247: undefined reference to `Polyhedron_demo_cut_plugin::enableAction()'
collect2: error: ld returned 1 exit status`

**Comment by maxGimeno** [comment]
This should work now.

**Comment by sloriot** [comment]
A file is missing when using the flat layout `/home/cgal_tester/build/src/cmake/platforms/ArchLinux-clang-CXX14/test/Polyhedron_Demo/Plugins/AABB_tree/Cut_plugin.cpp:7:10: fatal error: '../../AABB_tree/demo/AABB_tree/Color_ramp.h' file not found
#include "../../AABB_tree/demo/AABB_tree/Color_ramp.h"`

File: Polyhedron/demo/Polyhedron/Plugins/AABB_tree/Cut_plugin.cpp [hunk 3]
```diff
@@ -4,7 +4,8 @@
 #include <CGAL/AABB_intersections.h>
 
 #include "Scene.h"
-#include "../../AABB_tree/demo/AABB_tree/Color_ramp.h"
+//#include "../../AABB_tree/demo/AABB_tree/Color_ramp.h"
```
Comment (sloriot) [inline Polyhedron/demo/Polyhedron/Plugins/AABB_tree/Cut_plugin.cpp]: remove comment

File: Polyhedron/demo/Polyhedron/Plugins/AABB_tree/Cut_plugin.cpp [hunk 4]
```diff
@@ -572,12 +565,12 @@ class Q_DECL_EXPORT Scene_aabb_plane_item : public Scene_plane_item
 
       const FT& d00 = m_distance_function[i][j].second;
       // determines grey level
-      unsigned int i00 = 255-(unsigned)(255.0 * (double)std::fabs(d00) / m_max_distance_function);
+      double i00 = (double)std::fabs(d00) / m_max_distance_function;
```
Comment (sloriot) [inline Polyhedron/demo/Polyhedron/Plugins/AABB_tree/Cut_plugin.cpp]: double vs. FT

**Comment by sloriot** [comment]
There is a huge performance gap between the AABB-tree demo and the Polyhedron demo while moving the cutting plane.
About the display, the cutting plane in the demo seems to no be enlighted (colors are like in the shadow)+the grid should not be displayed.

**Comment by maxGimeno** [comment]
All is fixed, but it will need #1235 to work.

**Comment by maxGimeno** [comment]
It now fixes #1228

File: AABB_tree/include/CGAL/internal/AABB_tree/Halfedge_and_face_graph_property_maps.h [hunk 5]
```diff
@@ -192,7 +192,9 @@ struct Source_point_from_edge_descriptor{
   get(Source_point_from_edge_descriptor<HalfedgeGraph,VertexPointPMap> pmap,
       key_type h)
   {
-    return get(pmap.m_vppm,  source(h, *pmap.m_graph) );
+    return  get(vertex_point,
+                       *pmap.m_graph,
+                       source(h, *pmap.m_graph) );
```
Comment (sloriot) [inline AABB_tree/include/CGAL/internal/AABB_tree/Halfedge_and_face_graph_property_maps.h]: I guess you did not want to commit that change

**Comment by sloriot** [comment]
Branch was rebased to removed an extra commit

**Pull request merged**